src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -955,33 +955,96 @@ private IEnumerable<Expression> CompileInvocationArguments(IEnumerable<Expressio
             return arguments == null ? Array.Empty<Expression>() : arguments.Select(CompileExpressionOperand);
         }
 
-        internal Expression ReduceAssignment(ISupportsAssignment left, TokenKind tokenKind, Expression right)
+        internal Expression ReduceAssignment(
+            ISupportsAssignment left,
+            TokenKind tokenKind,
+            Expression right,
+            MemberExpressionAst lhsMemberAccessAst)
         {
-            IAssignableValue av = left.GetAssignableValue();
-            ExpressionType et = ExpressionType.Extension;
+            IAssignableValue assignmentTarget = left.GetAssignableValue();
+            var expressionType = ExpressionType.Extension;
 
-            switch (tokenKind)
+            if (tokenKind == TokenKind.Equals)
             {
-                case TokenKind.Equals: return av.SetValue(this, right);
-                case TokenKind.PlusEquals: et = ExpressionType.Add; break;
-                case TokenKind.MinusEquals: et = ExpressionType.Subtract; break;
-                case TokenKind.MultiplyEquals: et = ExpressionType.Multiply; break;
-                case TokenKind.DivideEquals: et = ExpressionType.Divide; break;
-                case TokenKind.RemainderEquals: et = ExpressionType.Modulo; break;
-                case TokenKind.QuestionQuestionEquals: et = ExpressionType.Coalesce; break;
+                return assignmentTarget.SetValue(this, right);
             }
 
+            expressionType = tokenKind switch
+            {
+                TokenKind.PlusEquals => ExpressionType.Add,
+                TokenKind.MinusEquals => ExpressionType.Subtract,
+                TokenKind.MultiplyEquals => ExpressionType.Multiply,
+                TokenKind.DivideEquals => ExpressionType.Divide,
+                TokenKind.RemainderEquals => ExpressionType.Modulo,
+                TokenKind.QuestionQuestionEquals => ExpressionType.Coalesce,
+                _ => throw new ArgumentException(nameof(tokenKind))
```

Comment:
This changes original behavior.

Reply:
Yeah, you're right. I can just move the variable initialization down to here and have the default be `ExpressionType.Extension`, I missed that. Thanks!

Reply:
It seems original default is null.